packages/web/src/components/cell/createSuspendingCell.tsx

```diff
@@ -0,0 +1,133 @@
+import { Suspense } from 'react'
+
+// @TODO(STREAMING): We are directly importing from Apollo here
+// because useBgQuery, and useReadQuery are Apollo 3.8+ specific
+import { NetworkStatus, QueryReference, useApolloClient } from '@apollo/client'
+
+import { useBackgroundQuery, useReadQuery } from '../GraphQLHooksProvider'
+
+/**
+ * This is part of how we let users swap out their GraphQL client while staying compatible with Cells.
+ */
+import { CellErrorBoundary } from './CellErrorBoundary'
+import {
+  SuspenseCellQueryResult,
+  CreateCellProps,
+  DataObject,
+  SuperSuccessProps,
+} from './cellTypes'
+import { isDataEmpty } from './isCellEmpty'
+
+/**
+ * Creates a Cell ~~ with Apollo Client only ~~
+ * using the hooks useBackgroundQuery and useReadQuery
+ */
+export function createSuspendingCell<
+  CellProps extends Record<string, unknown>,
+  CellVariables extends Record<string, unknown>
+>({
+  QUERY,
+  beforeQuery = (props) => ({
+    // By default, we assume that the props are the gql-variables.
+    variables: props as unknown as CellVariables,
+    /**
+     * We're duplicating these props here due to a suspected bug in Apollo Client v3.5.4
+     * (it doesn't seem to be respecting `defaultOptions` in `RedwoodApolloProvider`.)
+     *
+     * @see {@link https://github.com/apollographql/apollo-client/issues/9105}
+     */
+    fetchPolicy: 'cache-and-network',
+    notifyOnNetworkStatusChange: true,
+  }),
+  afterQuery = (data) => ({ ...data }),
+  isEmpty = isDataEmpty,
+  Loading = () => <>Loading...</>,
+  Failure,
+  Empty,
+  Success,
+  displayName = 'Cell',
+}: CreateCellProps<CellProps, CellVariables>): React.FC<CellProps> {
+  function SuperSuccess(props: SuperSuccessProps) {
+    const { RWJS_cellQueryRef, suspenseQueryResult, userProps } = props
+    const { data, networkStatus } = useReadQuery<DataObject>(RWJS_cellQueryRef)
+    const afterQueryData = afterQuery(data as DataObject)
+
+    const queryResultWithNetworkStatus = {
+      ...suspenseQueryResult,
+      networkStatus,
+    }
+
+    if (isEmpty(data, { isDataEmpty }) && Empty) {
+      return (
+        // @TODO HELP!
```

Comment:
The TS errors are too vague for me to understand how to fix this atm! 

Need a second pair of eyes

Reply:
Thanks @KrisCoulson